x-pack/plugins/fleet/public/components/uninstall_command_flyout/uninstall_command_flyout.test.tsx

```diff
@@ -58,43 +72,89 @@ describe('UninstallCommandFlyout', () => {
     });
   });
 
-  describe('rendering commands only for received platforms', () => {
-    it('renders commands for e.g. Linux and Mac', () => {
-      useCommandsMock.mockReturnValue({
-        linux: 'command for linux',
-        mac: 'command for mac',
-      });
-
+  describe('when successfully fetching uninstall tokens', () => {
+    it('renders buttons for Linux/Mac and for Windows', () => {
       const renderResult = render();
 
       const platformsButtonGroup = renderResult.getByTestId(
         'uninstall-commands-flyout-platforms-btn-group'
       );
-      expect(platformsButtonGroup).toHaveTextContent('Mac');
-      expect(platformsButtonGroup).toHaveTextContent('Linux');
-      expect(platformsButtonGroup).not.toHaveTextContent('Windows');
-      expect(platformsButtonGroup).not.toHaveTextContent('RPM');
-      expect(platformsButtonGroup).not.toHaveTextContent('DEB');
+      expect(platformsButtonGroup).toHaveTextContent('Linux or Mac');
+      expect(platformsButtonGroup).toHaveTextContent('Windows');
     });
 
-    it('renders commands for e.g. Mac, Windows, DEB and RPM', () => {
-      useCommandsMock.mockReturnValue({
-        mac: 'command for mac',
-        deb: 'commands for deb',
-        windows: 'commands for windows',
-        rpm: 'commands for rpm',
-      });
+    it('renders commands for Linux/Mac on default', () => {
+      const renderResult = render();
 
+      const uninstallInstructions = renderResult.getByTestId(
+        'uninstall-commands-flyout-code-block'
+      );
+      expect(uninstallInstructions).toHaveTextContent(
+        'sudo elastic-agent uninstall --uninstall-token 123456789'
+      );
+    });
+
+    it('when user selects Windows, it renders commands for Windows', () => {
       const renderResult = render();
 
-      const platformsButtonGroup = renderResult.getByTestId(
-        'uninstall-commands-flyout-platforms-btn-group'
+      renderResult.getByTestId('windows').click();
+
+      const uninstallInstructions = renderResult.getByTestId(
+        'uninstall-commands-flyout-code-block'
       );
-      expect(platformsButtonGroup).toHaveTextContent('Mac');
-      expect(platformsButtonGroup).toHaveTextContent('Windows');
-      expect(platformsButtonGroup).toHaveTextContent('RPM');
-      expect(platformsButtonGroup).toHaveTextContent('DEB');
-      expect(platformsButtonGroup).not.toHaveTextContent('Linux');
+      expect(uninstallInstructions).toHaveTextContent(
+        'C:\\"Program Files"\\Elastic\\Agent\\elastic-agent.exe uninstall --uninstall-token 123456789'
+      );
+    });
+  });
+
+  describe('fetching', () => {
```

Comment:
could be this a bit more specific? For example: `When loading and when error` or something similar. Or maybe you can move the loading `it` to the `describe` above and use this only for error cases.

Reply:
sure, thanks for the suggestion!
a241b63a720